### ISSUE TITLE
[BUG] PR #3061 breaks zip importing with "ImportError: The 'jaraco' package is required;"

### ISSUE THREAD

**Issue by asottile** [pypa/setuptools#3084]
### setuptools version

60.7.1

### Python version

3.8.10

### OS

ubuntu 20.04

### Additional environment information

n/a

### Description

this breaks virtualenv (and others as it seems from nuitka/nuitka#1406) https://github.com/pypa/virtualenv/runs/5077580857?check_suite_focus=true

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "<frozen zipimport>", line 259, in load_module
  File "/tmp/pytest-of-asottile/pytest-1/test_zip_importer_can_import_s0/site-packages.zip/setuptools/__init__.py", line 16, in <module>
  File "<frozen zipimport>", line 259, in load_module
  File "/tmp/pytest-of-asottile/pytest-1/test_zip_importer_can_import_s0/site-packages.zip/setuptools/version.py", line 1, in <module>
  File "<frozen zipimport>", line 259, in load_module
  File "/tmp/pytest-of-asottile/pytest-1/test_zip_importer_can_import_s0/site-packages.zip/pkg_resources/__init__.py", line 74, in <module>
  File "/tmp/pytest-of-asottile/pytest-1/test_zip_importer_can_import_s0/site-packages.zip/pkg_resources/extern/__init__.py", line 52, in create_module
  File "/tmp/pytest-of-asottile/pytest-1/test_zip_importer_can_import_s0/site-packages.zip/pkg_resources/extern/__init__.py", line 44, in load_module
ImportError: The 'jaraco' package is required; normally this is bundled with this package so if you get this warning, consult the packager of your distribution.
```

via this subprocess call

```
E           subprocess.CalledProcessError: Command '['/tmp/pytest-of-asottile/pytest-1/test_zip_importer_can_import_s0/venv/bin/python', '-c', 'from setuptools.dist import Distribution']' returned non-zero exit status 1.
```

note that this is exercising the zipimporter, so this likely breaks zip importing

### Expected behavior

not crashing

### How to Reproduce

```console
$ PYTHONPATH=setuptools-60.7.1-py3-none-any.whl python -c 'from setuptools import dist'
```

### Output

```console
$ PYTHONPATH=setuptools-60.7.1-py3-none-any.whl python -c 'from setuptools import dist'
Error processing line 1 of /tmp/y/virtualenv/venv/lib/python3.8/site-packages/distutils-precedence.pth:

  Traceback (most recent call last):
    File "/usr/lib/python3.8/site.py", line 175, in addpackage
      exec(line)
    File "<string>", line 1, in <module>
  AttributeError: module '_distutils_hack' has no attribute 'ensure_shim'

Remainder of file ignored
Error processing line 1 of /tmp/y/virtualenv/venv/lib/python3.8/site-packages/distutils-precedence.pth:

  Traceback (most recent call last):
    File "/usr/lib/python3.8/site.py", line 175, in addpackage
      exec(line)
    File "<string>", line 1, in <module>
  AttributeError: module '_distutils_hack' has no attribute 'ensure_shim'

Remainder of file ignored
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "<frozen zipimport>", line 259, in load_module
  File "/tmp/y/virtualenv/setuptools-60.7.1-py3-none-any.whl/setuptools/__init__.py", line 16, in <module>
  File "<frozen zipimport>", line 259, in load_module
  File "/tmp/y/virtualenv/setuptools-60.7.1-py3-none-any.whl/setuptools/version.py", line 1, in <module>
  File "<frozen zipimport>", line 259, in load_module
  File "/tmp/y/virtualenv/setuptools-60.7.1-py3-none-any.whl/pkg_resources/__init__.py", line 74, in <module>
  File "/tmp/y/virtualenv/setuptools-60.7.1-py3-none-any.whl/pkg_resources/extern/__init__.py", line 52, in create_module
  File "/tmp/y/virtualenv/setuptools-60.7.1-py3-none-any.whl/pkg_resources/extern/__init__.py", line 44, in load_module
ImportError: The 'jaraco' package is required; normally this is bundled with this package so if you get this warning, consult the packager of your distribution.
```

**Comment by jaraco** [comment]
I suspect the issue here is that `jaraco` is a namespace package and that the zip importer doesn't support namespace packages in subpackages but the standard importer does, and all the tests rely on the standard importer. Should be an easy fix.

**Comment by jaraco** [comment]
Fix releasing with 60.8.1.
